### PR TITLE
Allow subform field value to be an array

### PR DESCRIPTION
**Pull request by JamesNGarrett**
Pull Request for Issue #39768

### Summary of Changes

Allow subform field value to be an array when prepared for output.  The PlgFieldsSubform->onCustomFieldsBeforePrepareField() method assumes that the subform data will be coming from the database as json, whereas the data (for contact.mail custom fields) is coming from direct from the input passed to the FieldHelpers $valuesToOverride param as an array.  Hence, the change allows the json_decode to be skipped if the value is already an array.  

Also removed the redundant comment ("Check if the field should be processed by us") which has already been removed from some of the other field plugins.

### Testing Instructions

1. Go to Admin > Components > Contacts > Fields and switch the context select to Mail (ie. com_contact.mail)
2. Create a standard field, eg. text, within the mail context (set it to subform just for normalcy)
3. Create a subform with the standard field within the mail context
4. Add a contact
5. Add a single contact menu item to the front end and select the contact
6. Go to the menu item in the front end with a user that has permission to edit custom field values, eg. super admin (you can also set the field permissions to allow other groups to edit custom field values)
7. Submit the contact form with any values in the subform

### Actual result BEFORE applying this Pull Request

500 Error:  json_decode(): Argument #1 ($json) must be of type string, array given
Location: JROOT/plugins/fields/subform/subform.php:128

### Expected result AFTER applying this Pull Request

No error. The subform values are included in the email to the contact as html.

### CODE REVIEW

File: plugins/fields/subform/subform.php [hunk 1]
```diff
@@ -117,11 +117,14 @@ public function onContentPrepareForm(Form $form, $data)
      */
     public function onCustomFieldsBeforePrepareField($context, $item, $field)
     {
-        // Check if the field should be processed by us
```
Comment (richard67) [inline plugins/fields/subform/subform.php:120]: Why do you remove this comment?

Comment (richard67) [reply plugins/fields/subform/subform.php:120]: Ah ok you described that, I just see.

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on ae840fc2cf3fbc06171991413264f7b8cb3ca1c3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39842">issues.joomla.org/tracker/joomla-cms/39842</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on ae840fc2cf3fbc06171991413264f7b8cb3ca1c3<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39842">issues.joomla.org/tracker/joomla-cms/39842</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/39842">issues.joomla.org/tracker/joomla-cms/39842</a>.</sub>

**Comment by fancyFranci** [comment]
Thanks a lot!

**Pull request merged**